### PR TITLE
Unv output

### PR DESCRIPTION
**Pull request by roigcarlo**
This expands the work done by @Hubertus248 for the unv export.
I've added a process to use from python and the nodal results.

Please be aware that's its a very incipient implementation and not every kratos variable is supported.

### CODE REVIEW

**Comment by dbaumgaertner** [comment]
Just adding that at some point @KratosMultiphysics/optimization also needed such a unv-export and implemented it without generality in mind under applications/ShapeOptimizationApplication/custom_utilities/input_output/universal_file_io.h . But the implementation in this PR appears to be already more elaborate.

**Comment by loumalouomega** [comment]
@roigcarlo I added the license header to avoid potential "legal" problems.

**Comment by roigcarlo** [comment]
@loumalouomega Thx!
@dbaumgaertner Thx for pointing that out. I was not aware that Kratos already had one. I can extend it so the headers are customizable etc... This one is also a little bit hardcoded, but if I am not the only one to use it its worth the effort.

**Comment by philbucher** [comment]
Nice work!

- There is some commented code, do you plan to leave it there?
- Can you please add a test? There is currently a discussion going on abt unifying the output-processes (#3766) and it would be a shame breaking this when doing some refactoring

File: kratos/input_output/unv_output.cpp [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+}
+
+void UnvOutput::WriteNodes() {
+    std::ofstream outputFile;
+    outputFile.open(mOutputFileName, std::ios::out | std::ios::app);
+
+    outputFile << std::scientific;
+    outputFile << std::setprecision(15);
+
+    const int dataSetNumberForNodes = 2411;
```
Comment (loumalouomega) [inline kratos/input_output/unv_output.cpp]: This value?

File: kratos/input_output/unv_output.cpp [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+    outputFile.close();
+}
+
+
+void UnvOutput::WriteElements() {
+    std::ofstream outputFile;
+    outputFile.open(mOutputFileName, std::ios::out | std::ios::app);
+
+    const int dataSetNumberForElements = 2412;
```
Comment (loumalouomega) [inline kratos/input_output/unv_output.cpp]: This value?

File: kratos/input_output/unv_output.h [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+        outputFile << std::setw(6)  << "-1" << "\n";                                                // Begin block
+        outputFile << std::setw(6)  << dataSetNumberForResults << "\n";                             // DatasetID
+
+        outputFile << std::setw(10) << dataSetLabel << "\n";                                        // Record 1 - Label
+        outputFile << std::setw(6)  << dataSetName << "\n";                                         // Record 2 - Name
+        outputFile << std::setw(10) << as_integer(DatasetLocation::DATA_AT_NODES) << "\n";          // Record 3
+
+        // String records, seems like you can put anything you want.
+        outputFile << "Test1" << "\n";                                                               // Record 4
```
Comment (loumalouomega) [inline kratos/input_output/unv_output.h]: This should be removed?

**Review by loumalouomega — COMMENTED** [review]
Some comments

File: kratos/input_output/unv_output.h (hunk 3 above)
Comment (roigcarlo) [reply kratos/input_output/unv_output.h]: No, it has to be something, I can change it so its only a "\n"

File: kratos/input_output/unv_output.cpp (hunk 1 above)
Comment (roigcarlo) [reply kratos/input_output/unv_output.cpp]: This is the id of the dataset. 2411 -> nodes, 2412 -> Elements, 2414 -> Results,

I will move it a enum as well.

File: kratos/input_output/unv_output.cpp (hunk 2 above)
Comment (roigcarlo) [reply kratos/input_output/unv_output.cpp]: Same.

**Comment by roigcarlo** [comment]
Hi @philbucher Thx for the comments, i will look into the discussion for the unified output. 
I plan to comment it properly but we are in a bit of a hurry and this needs to be merged before Wednesday so right now are there so I don't forget what field does what.
Pretty much same for the tests, but they are planned.

**Comment by dbaumgaertner** [comment]
@roigcarlo Extending this unv output to include whats there and missing from our side (possibly there is even nothing) for sure is worth the effort on the long run. But now that you are a bit in a hurry, we can just wait for the merge, try it then and see, whether some functionality is missing from our point of view. After that, we could speak again regarding a possible extension.

**Comment by roigcarlo** [comment]
So test are passing now, someone can approve pls?

**Review by loumalouomega — APPROVED** [review]
imagen

**Pull request merged**